app/api/v1/devices.py

```
@@ -0,0 +1,120 @@
+            detail="Hardware already registered - reset the vault first",
+        )
+
+
+@router.post(
+    "/verify-pin",
+    response_model=VerifyPINResponse,
+    status_code=status.HTTP_200_OK,
+)
+async def verify_pin(
+    payload: VerifyPINRequest,
+    vault_repo: VaultRepository = Depends(get_vault_repo),
```

Ruff (0.15.2)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)